src/qtils/strict_sptr.hpp

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright Quadrivium LLC
+ * All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <memory>
+#include <stdexcept>
+#include <type_traits>
+
+namespace qtils {
+
+  /**
+   * @brief A strict version of std::shared_ptr that guarantees non-null values.
+   *
+   * This class prevents default construction and ensures that the shared
+   * pointer is never null. It removes unsafe methods such as `reset()` and
+   * `operator=`. Provides safe access to the underlying std::shared_ptr.
+   *
+   * @tparam T The type of the managed object.
+   */
+  template <class T>
+    requires(not std::is_void_v<T>)
+  class StrictSharedPtr {
+   public:
+    // Default constructor is deleted to prevent an uninitialized state
+    StrictSharedPtr() = delete;
+
+    // Explicitly delete constructor from nullptr
+    StrictSharedPtr(std::nullptr_t) = delete;
+
+    /**
+     * @brief Constructs a StrictSharedPtr from an existing std::shared_ptr.
+     *
+     * @param sptr The shared pointer to be wrapped.
+     * @throws std::invalid_argument if the pointer is null.
+     */
+    // NOLINTNEXTLINE(google-explicit-constructor)
+    StrictSharedPtr(std::shared_ptr<T> sptr) : ptr_(std::move(sptr)) {
+      if (!ptr_) {
+        throw std::invalid_argument("Shared pointer is null");
```

Comment:
Hm, we mostly avoid exceptions in our projects, so I'd like to have a static method that returns outcome::result or std::expected along with this, I think? Also, we may as well add a constructor to initialize it from a raw pointer. And something that duplicates make_shared functional. While I'm here, I'd like to propose the name SharedRef again, as it is shorter and matches existing  C++ dictionary.

Reply:
`StrictSharedPtr` doesn't throw if used correctly.
It throws when there is bug in calling code,
because throwing error shows exact location in debugger or stack trace, while outcome/expected hide source of error.

Reply:
With that logic assert should be used instead of throwing, or just explicit call to abort() to avoid removing it in release build.